src/analysis/complex/conformal.lean

```diff
@@ -0,0 +1,141 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import analysis.complex.isometry
+import analysis.calculus.conformal
+
+/-!
+# Complex Conformal Maps
+
+We prove the sufficient and necessary conditions for a complex function to be conformal.
+
+## Main results
+
+* `is_complex_or_conj_complex_linear_iff_is_conformal_map`: the main theorem for linear maps.
+* `conformal_at_iff_holomorphic_or_antiholomorph`: the main theorem for complex functions.
+
+## Warning
+
+Antiholomorphic functions such as the complex conjugate are considered as conformal functions in
+this file.
+-/
+
+noncomputable theory
+
+section complex_conformal_linear_map
+
+variables {f : ℂ → ℂ} {z : ℂ} {g : ℂ →L[ℝ] ℂ}
+
+open complex linear_isometry linear_isometry_equiv continuous_linear_map
+     finite_dimensional linear_map
+
+lemma is_complex_linear_iff_holomorph (hf : differentiable_at ℝ f z) :
+  differentiable_at ℂ f z ↔ is_linear_map ℂ (fderiv ℝ f z) :=
```

Comment:
I'd prefer the statement: 
```
  differentiable_at ℂ f z ↔ ∃ h : ℂ →ₗ[ℂ]ℂ, h.restrict_scalars ℝ = fderiv ℝ f z
```
(and it should live in `analysis.calculus.fderiv`, phrased for two fields `𝕜`, `𝕜'` with `[normed_algebra 𝕜 𝕜']`, rather than just for `ℝ` and `ℂ`).

Reply:
Maybe preceded by a version for `differentiable_within_at 𝕜' f s z`, for a set `s` satisfying `unique_diff_within_at 𝕜 s z`.